rust/src/lib/ip.rs

```diff
@@ -851,6 +872,42 @@ impl From<InterfaceIpv6> for InterfaceIp {
     }
 }
 
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize,
+)]
+#[serde(rename_all = "lowercase")]
+#[non_exhaustive]
+pub enum Forwarding {
+    /// Use the NetworkManager global default value.
+    #[serde(rename = "default")]
+    Default = -1,
+    /// Disable forwarding on this interface.
+    #[serde(rename = "no", alias = "false")]
+    No = 0,
+    /// Enable forwarding on this interface.
+    #[serde(rename = "yes", alias = "true")]
+    Yes = 1,
+    /// Enable forwarding if any shared connection is active.
+    #[serde(rename = "auto")]
+    Auto = 2,
```

Comment:
Nmstate has no design of shared connection, so this `Forwarding` should be boolean.
Please not use NM design here.

Reply:
I changed the `forwarding` setting to boolean, thanks!